### PR TITLE
inductor: meta registration for mkldnn ops

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91299

Fix https://github.com/pytorch/torchdynamo/issues/198, which supports Meta tensor for conv/linear fused ops to reduce the compilation time.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -514,31 +526,123 @@ def pick_memory_format():
...
+        stride,
+        dilation,
+        groups,
+        binary_attr,
+        alpha,
+        unary_attr,
+        unary_scalars,
+        unary_algorithm,
+    ):
+        out = input_tensor.new_empty(other.size())
```
Comment (jgong5) [inline torch/_meta_registrations.py:597]: Shall we add an assertion that the size of `other` matches the one calculated from `calc_conv_nd_return_shape` to make sure other args are what we expect?

**Review by jgong5 — APPROVED** [review]
LGTM. Better to add some assertions to mkldnn fused ops. Also, do we have unit tests covering them?

File: torch/_meta_registrations.py (hunk 1 above)
Comment (XiaobingSuper) [reply torch/_meta_registrations.py:597]: I don't think we need such a check, we have made sure the size matches at inductor side, I don't think the user will directly call this path.

**Review by EikanWang — APPROVED** [review]

File: torch/_meta_registrations.py (hunk 1 above)
Comment (jgong5) [reply torch/_meta_registrations.py:597]: OK. How about adding a note here?

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**